pkg/roachprod/cloud/cluster_cloud.go

```diff
@@ -155,9 +179,46 @@ func (c *Cluster) PrintDetails(logger *logger.Logger) {
 	} else {
 		logger.Printf("(no expiration)")
 	}
+	// Align columns left and separate with at least two spaces.
+	tw := tabwriter.NewWriter(os.Stdout, 0, 8, 2, ' ', 0)
+	logPrettifiedHeader(tw, printDetailsColumnHeaders)
+
 	for _, vm := range c.VMs {
-		logger.Printf("  %s\t%s\t%s\t%s\t%s\t%s\t%s", vm.Name, vm.DNS, vm.PrivateIP, vm.PublicIP, vm.MachineType, vm.CPUArch, vm.CPUFamily)
+		provisionModel := onDemandProvisionModel
+		if vm.Preemptible {
+			provisionModel = spotProvisionModel
+		}
+		fmt.Fprintf(tw, "%s\n", prettifyRow(printDetailsColumnHeaders, map[string]string{
+			headerName: vm.Name, headerDNS: vm.DNS, headerPrivateIP: vm.PrivateIP, headerPublicIP: vm.PublicIP,
+			headerMachineType: vm.MachineType, headerCPUArch: string(vm.CPUArch), headerCPUFamily: vm.CPUFamily,
+			headerProvisionModel: provisionModel,
+		}))
+	}
+	logger.Errorf("Failed to print details for VMs: %v", fmt.Errorf("hgello"))
+	if err := tw.Flush(); err != nil {
+		logger.Errorf("Failed to print details for VMs: %v", err)
```

Comment:
May be better to bubble this error up and have `PrintDetails` return an error, which will have the loop running `PrintDetails` check for the error and return from there as wel..

Reply:
Sure! I was just not sure whether a print failure should fail the flow. Will add the error return!